resources/inputs/input.tsx

```diff
@@ -2,42 +2,94 @@ import React from "react";
 import { useTranslations } from "next-intl";
 import form_style from "./styles/input.module.css";
 
-interface AuthInputProps {
+interface InputFieldProps {
   register: any;
   errors: any;
+  name: string;
+  type: string;
+  placeholder: string;
+  validation: object;
 }
 
-const Input: React.FC<AuthInputProps> = ({ register, errors }) => {
+const InputField: React.FC<InputFieldProps> = ({
+  register,
+  errors,
+  name,
+  type,
+  placeholder,
+  validation,
+}) => {
   const t = useTranslations("");
 
+  return (
+    <div className={form_style.formInput}>
+      <input
+        {...register(name, validation)}
+        type={type}
+        placeholder={t(placeholder)}
+        className={`${form_style.formControl} ${form_style[`input${type.charAt(0).toUpperCase() + type.slice(1)}Icon`]}`}
+      />
+      {errors[name] && (
+        <span className={form_style.errorMessage}>
+          {t(errors[name].message)}
+        </span>
+      )}
+    </div>
+  );
+};
+
+interface AuthInputProps {
+  register: any;
```

Comment:
Dont put any, try to find out the type, of not the whole type at least the type of what we use